StrictMode and files under
/storage/sdcard1/
/storage/sdcard0/
android.os.FileUriExposedException: file:///storage/sdcard1/Downloads/taskmanager.ver.2.8.2.4.apk
at android.os.StrictMode.onFileUriExposed(StrictMode.java:1799)
at android.net.Uri.checkFileUriExposed(Uri.java:2346)

diff --git a/core/java/android/net/Uri.java b/core/java/android/net/Uri.java
index f76a683..ad550e8 100644
--- a/core/java/android/net/Uri.java
+++ b/core/java/android/net/Uri.java
@@ -2350,7 +2350,8 @@ public Uri getCanonicalUri() {
      * @hide
      */
     public void checkFileUriExposed(String location) {
-        if ("file".equals(getScheme()) && !getPath().startsWith("/system/")) {
+        if ("file".equals(getScheme()) && !(getPath().startsWith("/system/")
+                || getPath().startsWith("/storage/") || getPath().startsWith("/data/"))) {
             StrictMode.onFileUriExposed(this, location);
         }
     }
